### DISCUSSION TITLE
Benefits of Using ScriptKit vs Raycast Node scripts

### DISCUSSION THREAD

**Discussion by Jaballadares** [johnlindquist/kit#730] [General]
Hi there! I have been interested in using ScriptKit for awhile now, but feel like I may be missing something. Is there something ScriptKit enables that Raycast currently does not? 

I just downloaded ScriptKit and will use it and Raycast in tandem Super excited and thank you for creating such a valuable tool!

The noticeable difference in app size between the two prompted me to ask.

**Comment by johnlindquist** [comment]
Hi @Jaballadares , great question! While you'll see a lot of similarities, I think the big difference is the focus of each app:

## Script Kit's focus is making scripts easier to write, run, and share.

## Write
- You can create a script just by typing the name of it from the main menu
- Script Kit has a built-in editor
- Script Kit can auto-install packages from npm (if the user confirm's they trust it)
- Script Kit ships with a simple API/SDK to help you get your script done

## Run
- The scripts can also be run in the terminal
- You can run scripts with keyboard shortcuts
- A script can run another script with `await run("my-other-script")`
- You can run Script Kit scripts from Raycast/Alfred/Keyboard Maestro/Karabiner/whatever
- Eventually, it will be easy to run scripts from webhooks, GitHub actions, and anywhere else. (This is currently possible, just takes a bit of setup that I want to make easier)
- Script Kit will soon have a built-in terminal

## Share
- Scripts are copy/pastable
- Scripts can be installed instantly by clicking a link. [Open instantly-shared-script in Script Kit](https://scriptkit.com/api/new?name=instantly-shared-script&url=https://gist.githubusercontent.com/johnlindquist/0c13dc84ec8f5883f3ae367b74820031/raw/d40116adba9a4e2bbe36523dfc4548c103e0d404/instantly-shared-script.ts")

```js
// Name: Instantly Shared Script

import "@johnlindquist/kit"

await arg("I made this in 5 seconds")
```

- You can see the content of the script from the main ui
- You can read the script and know what it does, for example this is a script:
```js
let message = await arg("Type something to speak")
say(message)
```
- There's a community of shared scripts here: https://www.scriptkit.com/scripts. Everyone who shares a script gets their own page
- Eventually, scripts will be shared across Mac, Windows, and Linux (Windows and Linux are in dev previews)

I think Raycast is a great product with a different focus.

**Reply by Jaballadares** [reply]
Forgive the extremely delayed response!
Thank you for responding so thoroughly - the fact ScriptKit scripts are flexible and can essentially be triggered from anywhere really helps distinguish it from Raycast. 

I think this may be my gateway into some system level automations. 

Thank you again!
